street_level_crime/input_helpers.py

```diff
@@ -1,14 +1,66 @@
-import re
-
-uk_postcode_format = re.compile("([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})")
-
-def validate_postcode(postcode):
-    return bool(uk_postcode_format.match(postcode))
-
-def ui_get_postcode():
-    while True:
-        postcode = input("Please, enter postcode (e.g. BH12 2ED)")
-        if validate_postcode(postcode):
-            print ("Correct postcode")
-        else:
-            print("Please, check your postcode. It must be a valid UK format (e.g. GH12 2ED)")
+import re
+
+uk_postcode_format = re.compile("([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})")
+valid_file_name = re.compile('[^\\\<>:"/|?*]')
+
+def validate_postcode(postcode):
+    return bool(uk_postcode_format.match(postcode))
+
+def validate_file_name(file_name):
+    return bool(valid_file_name.match(file_name))
+
+def ui_get_postcode():
+    while True:
+        postcode = input("Please, enter postcode (e.g. BH12 2ED)")
+        if validate_postcode(postcode):
+            break
+        else:
+            print("Please, check your postcode. It must be a valid UK format (e.g. GH12 2ED)")
+
+def ui_get_distance():
+    while True:
+        distance = input("Please, select distance from postcode centre: [1] 1km \n [2] 2km \n [3] 5km \n")
+        if distance == str(1) or option == str(2) or option == str(3):

```

Comment:
I like this, by comparing to the string version of the number you don't have to worry about a type cast error. I've just rewritten it to be a little shorter, we can write the string literal instead of doing a type conversion as we know what the result will be. By having an array it's easier to add more options and I think it's a bit shorter
```suggestion
        if distance in ["1", "2", "3"]:
```